### DISCUSSION TITLE
Setup Neotest for python

### DISCUSSION THREAD

**Discussion by msMatix** [LazyVim/LazyVim#1856] [Q&A] [answered]
Hey everyone :),

I just wanted to use neotest in combination with python.
For this I have enabled neotest under extras and also use the extra for python language. 

Then I go into an empty project with a test file and start the tests. Unfortunately, I get the message: no tests found.

Folder structure:
```
- my_neotest_project
      - test_neo.py
```

test_neo.py:
```
import pytest

def test_hello():
    assert 1 == 1

def test_hello2():
    assert 2 == 1
```

When I open Summary with `<leader>ts` I see the files and also executing with `<leader>tt` seems to work. But as soon as I do Show Output `<leader>to` I get the mentioned warning.

Can someone help me with this?
Do I have enventuell except the enablen of the plugin other things to configure?

Thanks in advance.

Matix

**Comment by dpetka2001** [comment]
So in the `neotest-python` docs it states, that if `pytest` is available it will be used by default, but I also couldn't get your test to run, so I defined `runner = "pytest"` and seemed to work. Try the following in a file in `lua/plugins/` directory
```lua
  {
    "nvim-neotest/neotest",
    dependencies = {
      "nvim-neotest/neotest-python",
    },
    opts = {
      adapters = {
        ["neotest-python"] = {
          -- Here you can specify the settings for the adapter, i.e.
          runner = "pytest",
          -- python = ".venv/bin/python",
        },
      },
    },
  },
```

**Reply by msMatix** [reply]
unfortunately this did not work for me. i also disabled all my own plugins and only installed the extras that are needed. 
Did you use a venv or similar?

**Reply by dpetka2001** [reply]
Yes I checked in a venv in my setup.

**Reply by msMatix** [reply]
Ok I have now added the venv and then activated it. then restarted the tests. 
Unfortunately the warning remains. Have you adjusted anything else?

**Reply by msMatix** [reply]
you only added pytest to the venv right? so `pip3 install pytest`

**Reply by dpetka2001** [reply]
Yes, nothing else. My neotest setup is the above that I posted.

**Reply by msMatix** [reply]
I see. I just tried a complete new LazyVim installation and then enabled extra for python and for neotest. In addition to that I also used your config for neotest. Unfortunately that does not work as well. 

So it may be another problem...

**Reply by dpetka2001** [reply]
Someone opened a new PR that added treesitter as a dependency to neotest. See, if you do that if it maybe solves your problem, although I didn't have to add treesitter as dependency for it to work.

**Reply by msMatix** [reply, marked as answer]
ah great tipp!
With that hint everything worked.
Here is my solution:

```
return {
  {
    "nvim-neotest/neotest",
    dependencies = {
      "nvim-neotest/neotest-python",
      "nvim-treesitter/nvim-treesitter",
    },
  },
}
```

**Reply by dpetka2001** [reply]
I wonder why that is needed and I don't need it in my case. Very strange.

**Reply by dpetka2001** [reply]
Are you using Neovim stable version?

**Reply by msMatix** [reply]
yes I use a stable version
Could it be because I am still adding the extra plugins in the old variant and not yet using the new command :LazyExtras?
In the docs I read something about the correct order?

**Reply by dpetka2001** [reply]
I just tried it with Neovim stable and the treesitter dependency is needed indeed. I was using Neovim nightly and I didn't have any problem without that dependency. I'm not sure what's the problem between Neovim versions.

**Reply by msMatix** [reply]
that is interesting. thank you for the hint
